docs/usage.md

```diff
@@ -30,74 +36,92 @@ For more specific integration options, please check out the [integration documen
 
-- [Generate from a pure JS object](../examples/swagger2.0-from-object)
+- [Generate from a Swagger 2.0 JS object](../examples/swagger2.0-from-object)
 
 The Swagger input processor expects that the property `swagger` is defined in order to know it should be processed.
 
 The response payload and `body` parameters, since it is a JSON Schema variant, is [interpreted as a such](./interpretation_of_JSON_Schema.md).
 
+### Limitations and Compatibility
+These are the current known limitation of the Swagger 2.0 input.
+
+#### Polymorphism
+
+Through the Swagger 2.0 Schema you are able to use `discriminator` for achieving polymorphism. Current version of Modelina does not generate the expected inheritance of models, instead it's current behavior is to [merge the schemas together](./inputs/interpretation_of_JSON_Schema.md#Processing-sub-schemas). This means you will still get access to the properties that missing inherited model has, but without the inheritance.
+
+Long term if this is something you wish was supported, voice your [opionion here](https://github.com/asyncapi/modelina/issues/108).
+
 ## Generate models from OpenAPI documents
 
 There are one way to generate models from an OpenAPI document
 
-- [Generate from a pure JS object](../examples/openapi-from-object)
+- [Generate from OpenAPI 3.0 JS object](../examples/openapi-from-object)
 
 The OpenAPI input processor expects that the property `openapi` is defined in order to know it should be processed.
 
 The response and request payloads, since it is a JSON Schema variant, is [interpreted as a such](./interpretation_of_JSON_Schema.md).
 
+#### Limitations and Compatibility
+These are the current known limitation of the OpenAPI inputs.
+
+##### Polymorphism
+
+Through the OpenAPI 3.0 Schema you are able to use `discriminator` for achieving polymorphism. Current version of Modelina does not generate the expected inheritance of models, instead it's current behavior is to [merge the schemas together](./inputs/interpretation_of_JSON_Schema.md#Processing-sub-schemas). This means you will still get access to the properties that missing inherited model has, but without the inheritance.
+
+Long term if this is something you wish was supported, voice your [opionion here](https://github.com/asyncapi/modelina/issues/108).
+
 ## Generate models from TypeScript type files
 
 Currently, we support generating models from a TypeScript type file.
 
 - [Generate Java model from a TypeScript file](../examples/java-from-typescript-type/)
 
-The TypeScript input processor expects that the typescript file and base directory where it's present, is passed as input, in order to process the types accurately.
+The TypeScript input processor expects that the typescript file and base directory where it's present, is passed as input, in order to process the types accurately. The input processor converts the TypeScript types into JSON Schema, which is [interpreted as a such](./interpretation_of_JSON_Schema.md). 
```

Comment:
interpreted as a such? 😄

Reply:
Yeah, please, it's hard to deduce.

Reply:
Alright, I changed how we linked to the JSON Schema processor, let me know if it's sufficient :)